### ISSUE TITLE
SessionProvider sends many requests to retrieve the session

### ISSUE THREAD

**Issue by srinesha** [nextauthjs/next-auth#10244]
### Environment

```
  System:
    OS: Linux 6.1 NixOS 23.11 (Tapir) 23.11 (Tapir)
    CPU: (12) x64 12th Gen Intel(R) Core(TM) i7-1255U
    Memory: 8.62 GB / 15.34 GB
    Container: Yes
    Shell: 5.9 - /run/current-system/sw/bin/zsh
  Binaries:
    Node: 21.6.2 - ~/.nix-profile/bin/node
    npm: 10.2.4 - ~/.nix-profile/bin/npm
    pnpm: 8.15.3 - ~/.nix-profile/bin/pnpm
  Browsers:
    Chromium: 122.0.6261.94
  npmPackages:
    next: 14.1.2 => 14.1.2
    next-auth: 5.0.0-beta.15 => 5.0.0-beta.15
    react: ^18 => 18.2.0
```

### Reproduction URL

https://github.com/srinesha/nextjs-testing

### Describe the issue

I have setup credential provider authentication. When `SessionProvider` is added in the layout, one page reload sends around 5 requests to retrieve the session even though useSession is never used anywhere in the application.

SessionProvider in layout.tsx:
https://github.com/srinesha/nextjs-testing/blob/ec5d831cd926fd86f9f0b1952df91829034ba4c3/app/settings/layout.tsx#L10C1-L14C23

https://github.com/nextauthjs/next-auth/assets/160213066/6a1e2bc1-77fc-4159-9088-82b37bd4ded0

### How to reproduce

https://github.com/srinesha/nextjs-testing

1. Clone the repository
2. Install dependencies `pnpm install`
3. Run `pnpm dev`
4. Go to `localhost:3000/auth/login`
5. Enter any random email address
6. Click on submit
7. When loading the settings page, there will be many requests to get the session

### Expected behavior

Send one request per page load?

**Comment by muadpn** [comment]
<SessionProvider
      refetchOnWindowFocus={false}
      refetchWhenOffline={false}
      session={session}
    >
      {children}
    </SessionProvider>
    
    also if you open multiple window in the same browser you are using each window will ask for revalidation.

**Comment by abencun-symphony** [comment]
Same, no workaround helps, I always get three `/api/auth/session` calls - whenever the session refetch happens. Version `5.0.0-beta.16` on Next 14.14.

**Comment by SinPP** [comment]
Can confirm: three session requests per app load.

**Comment by abencun-symphony** [comment]
Still present in `beta.16`

**Comment by luchsamapparat** [comment]
I think this solves the issue at least partially: #10762

Another issue may be React's `StrictMode`

**Comment by luchsamapparat** [comment]
From what I understand, there are two bugs that cause unnecessary `/session` requests.

1. `broadcast().postMessage(...)` and `broadcast().addEventListener(...)` work on different `BroadcastChannel` instances within the same window, which means that messages that are intended for other windows are also handled by the same window. In consequence, every session update that is broadcasted also causes another session update within the same window.
2. When working in development mode, React runs every effect twice. The broadcast effect runs `broadcast().removeEventListener(...)` for cleaning up in between, but since `broadcast()` alwass returns a new `BroadcastChannel` instance, no listener is removed and we end up with two instances with listeners that call `getSession` for each session update.

I fixed both in this PR: https://github.com/nextauthjs/next-auth/pull/10762

**Comment by abencun-symphony** [comment]
I can confirm that the issue was resolved in `beta.18` for us.

**Comment by balazsorban44** [comment]
This was indeed fixed in #10762, great research and PR by @luchsamapparat! :green_heart:

**Comment by coderpan** [comment]
> I can confirm that the issue was resolved in `beta.18` for us.

After test, the issue was not resolved in beta.18

**Comment by nikuharuki** [comment]
Still not resolved for me

**Comment by ugniusramanauskas** [comment]
Same. Still not resolved in beta 21.

**Comment by IDJGILL** [comment]
Using 5.0.0-beta.25 and I confirm it is still not resolved.

**Comment by koodev2023** [comment]
5.0.0-beta.25, got 4

 GET / 200 in 305ms
 GET /favicon.ico 200 in 17ms
 GET /manifest.json 200 in 22ms
 GET /api/auth/session 200 in 24ms
 GET /api/auth/session 200 in 34ms
 GET /api/auth/session 200 in 16ms
 GET /api/auth/session 200 in 12ms

**Comment by L1teD** [comment]
```
✓ Compiled /api/auth/[...nextauth] in 410ms
 GET /api/auth/session 200 in 864ms
 GET /api/auth/session 200 in 684ms
 GET /api/auth/session 200 in 684ms
 GET /api/auth/session 200 in 53ms
 GET /api/auth/session 200 in 34ms
 GET /api/auth/session 200 in 39ms
 GET /api/auth/session 200 in 34ms
 GET /api/auth/session 200 in 33ms
 GET /api/auth/session 200 in 34ms
 GET /api/auth/session 200 in 45ms
 GET /api/auth/session 200 in 41ms
 GET /api/auth/session 200 in 39ms
 GET /api/auth/session 200 in 37ms
```
should be reopened

**Comment by NoahGdev** [comment]
reopen pls

**Comment by lfgtavora** [comment]
need reopen

**Comment by died14** [comment]
Not resolved at all.

**Comment by TexLuciano** [comment]
Not resolved at all, 5.0.0-beta.25  got 4

**Comment by 0-don** [comment]
still not resolved

**Comment by jlalmes** [comment]
🐛 still here: `v5.0.0-beta.29`

I always see two session requests when the page loads or is refocused.

<img width="613" alt="Image" src="https://github.com/user-attachments/assets/8f1d1cb8-073b-453b-a176-deffac2c9f1d" />

From looking into the bug & debugging, it seems that the first `__NEXTAUTH._getSession` call with `event: undefined` calls `await getSession({ broadcast: true })` which then triggers a `broadcast()` event, therefore triggering a new  `__NEXTAUTH._getSession` call with `event: "storage"` which forces call to `await getSession({ broadcast: false })` again.

The issue is that the `BroadcastChannel` is listening to events triggered by the current page, which was introduced here: https://github.com/nextauthjs/next-auth/pull/11470

@balazsorban44 can you reopen & I will open a PR to fix 🙏 


EDIT:

PR fix here: https://github.com/nextauthjs/next-auth/pull/13084

**Comment by evokelektrique** [comment]
Same issue, still not resolved

**Comment by jlalmes** [comment]
@evokelektrique, still waiting on PR https://github.com/nextauthjs/next-auth/pull/13084 to be merged :)

**Comment by TexLuciano** [comment]
@jlalmes two only you is my hero hahahah
